octobot_trading/signals/util.py

```diff
@@ -0,0 +1,69 @@
+#  Drakkar-Software OctoBot-Trading
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import octobot_trading.enums as trading_enums
+import octobot_trading.constants as trading_constants
+
+
+def get_signal_exchange_type(exchange_manager):
+    if exchange_manager.is_spot_only:
+        return trading_enums.TradingSignalExchangeTypes.SPOT
+    if exchange_manager.is_future:
+        return trading_enums.TradingSignalExchangeTypes.FUTURE
+    if exchange_manager.is_margin:
+        return trading_enums.TradingSignalExchangeTypes.MARGIN
+    return trading_enums.TradingSignalExchangeTypes.UNKNOWN
+
+
+def create_order_signal_description(
+        order, action,
+        target_amount=None,
+        target_position=None,
+        updated_quantity=trading_constants.ZERO,
+        updated_limit_price=trading_constants.ZERO,
+        updated_stop_price=trading_constants.ZERO,
+        updated_current_price=trading_constants.ZERO,
+    ):
+    # prevent circular import
+    import octobot_trading.personal_data as personal_data
+    return {
+        trading_enums.TradingSignalOrdersAttrs.ACTION.value: action.value,
+        trading_enums.TradingSignalOrdersAttrs.SIDE.value: order.side.value,
+        trading_enums.TradingSignalOrdersAttrs.TYPE.value:
```

Comment:
I don't understand why we need to pass the class name, I think we should prefer the TradeOrderType value

Reply:
Me too, the problem here is that we want to figure out the `TraderOrderType` of an order and to my knowledge we don't have a way to do that, that's why I added this `get_order_type_from_class` function

Reply:
nevermind, there is `order.order_type`